tests/e2e/specs/drawing.spec.js

```diff
@@ -0,0 +1,111 @@
+        goToDrawing()
+        clickTool('marker')
+        cy.get(olSelector).click(100, 100)
+        readGeoJsonType('Point')
+    })
+    it('create a text', () => {
+        goToDrawing()
+        clickTool('text')
+        cy.get(olSelector).click(100, 100)
+        readGeoJsonType('Point')
+    })
+    it('create a line with click first point', () => {
+        goToDrawing()
+        clickTool('line')
+        cy.get(olSelector).click(100, 100)
+        cy.get(olSelector).click(150, 100)
+        cy.get(olSelector).click(150, 150)
+        cy.get(olSelector).click(100, 100)
+        readGeoJsonType('Polygon')
+    })
+    it('create a line with double click', () => {
+        goToDrawing()
+        clickTool('line')
+        cy.get(olSelector).click(100, 100)
+        cy.get(olSelector).click(150, 100)
+        cy.get(olSelector).dblclick(150, 150)
+        readGeoJsonType('Polygon')
+    })
+    it('create a polygon', () => {
+        goToDrawing()
+        clickTool('line')
+        cy.get(olSelector).click(100, 100)
+        cy.get(olSelector).click(150, 100)
+        cy.get(olSelector).click(150, 150)
+        cy.get(olSelector).click(100, 100)
+        readGeoJsonType('Polygon')
+    })
+    it('test saving', () => {
+        const checkResponse = (interception, types, create = false) => {
+            if (!create) {
+                const urlArray = interception.request.url.split('/')
+                const id = urlArray[urlArray.length - 1]
+                expect(id).to.be.eq(mockResponse.adminId)
+            }
+            expect(interception.request.headers['content-type']).to.be.eq(
+                'application/vnd.google-earth.kml+xml'
+            )
+            expect(interception.request.body).to.be.contain('</kml>')
+            types.forEach((type) => expect(interception.request.body).to.be.contain(type))
+        }
+
+        cy.mockupBackendResponse('files', mockResponse, 'saveFile')
+        cy.mockupBackendResponse('files/**', { ...mockResponse, status: 'updated' }, 'modifyFile')
+
+        goToDrawing()
+        clickTool('marker')
+
+        cy.get(olSelector).dblclick('center')
+        cy.wait('@saveFile').then((interception) => checkResponse(interception, ['MARKER'], true))
+        cy.readStoreValue('state.drawing.drawingKmlIds').then((ids) => {
+            expect(ids.adminId).to.eq('test')
+            expect(ids.fileId).to.eq('test')
+        })
+
+        cy.get(olSelector).click('center')
+        cy.wait('@modifyFile').then((interception) => checkResponse(interception, ['MARKER']))
+        clickTool('line')
+        cy.get(olSelector).click(100, 100)
+        cy.get(olSelector).dblclick(150, 100)
+        cy.wait('@modifyFile').then((interception) =>
+            checkResponse(interception, ['MARKER', 'LINE'])
+        )
+        cy.readStoreValue('state.drawing.drawingKmlIds').then((ids) => {
+            expect(ids.adminId).to.eq('test')
+            expect(ids.fileId).to.eq('test')
```

Comment:
We could use the `mockResponse` object here instead of hardcoded value. This way if we change it above the test will still be valid